### DISCUSSION TITLE
Csla 6 Custom HttpProxy

### DISCUSSION THREAD

**Discussion by JacoJordaan** [MarimerLLC/csla#2656] [Questions]
In Csla 5 this works: `CslaConfiguration.Configure().DataPortal().DefaultProxy(typeof(SmartHttpProxy), "Local");`

How would configure a custom HttpProxy in Csla 6?

**Comment by rockfordlhotka** [comment]
In CSLA 6 the way to use the default HttpProxy is like this:

```c#
  services.AddCsla(o=>o
     .DataPortal()
       .UseHttpProxy(options => options.DataPortalUrl = "/api/DataPortal"));
```

For a custom proxy type, prior to calling `AddCsla`, you would (more or less) do this:

```c#
  services.AddTransient(typeof(IDataPortalProxy), typeof(MyCustomProxy));
  services.AddCsla();
```

**Reply by rockfordlhotka** [reply]
If you want to see how the `UseHttpProxy` extension method is implemented (thus allowing configuration values like the URL), you can see the code here:

https://github.com/MarimerLLC/csla/blob/main/Source/Csla/DataPortalClient/HttpProxyExtensions.cs

**Comment by GreatBarrier86** [comment]
I have a question on this. In CSLA 5, we had  the `Csla.ApplicationContext.DataPortalProxy` and `Csla.ApplicationContext.DataPortalUrlString` static properties that allowed changing the values at runtime (eg: switching from local to remote at an arbitrary time). 

In principle, it doesn't appear that Dependency Injection supports that concept. Would you say changing the proxy/url is simply not a thing we could do anymore?

**Reply by rockfordlhotka** [reply]
@GreatBarrier86 can you make this a new thread so it doesn't get lost?
